### PR TITLE
Changed tf_flowers to iterate over tgz instead of download_and_extract , issue #658

### PR DESCRIPTION
**Pull request by Suhasnama**
Changed `_split_generators()` and `_generate_examples()` to iterate over tgz instead of download_and_extract.

### CODE REVIEW

**Comment by Conchylicultor** [comment]
Thanks for your contribution. Please format the code following: https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style

The tests are currently failing:
```
    gen_kwargs={"archive": dl_manager.iter_archive(path)},
tensorflow_datasets/core/download/download_manager.py:323: in iter_archive
    return extractor.iter_archive(resource.path, resource.extract_method)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
path = '/tmpfs/src/github/datasets/tensorflow_datasets/testing/test_data/fake_examples/tf_flowers/flower_photos'
method = <ExtractMethod.NO_EXTRACT: 1>
    def iter_archive(path, method):
      """Yields (path_in_archive, f_obj) for archive at path using `tfds.download.ExtractMethod`."""  # pylint: disable=line-too-long
>     return _EXTRACT_METHODS[method](path)
E     KeyError: <ExtractMethod.NO_EXTRACT: 1>
tensorflow_datasets/core/download/extractor.py:200: KeyError
```
You also need to update the fake data.

**Comment by Suhasnama** [comment]
@Conchylicultor , thanks for reviewing , I had updated fake data i.e, added [flowers_photos.zip](https://github.com/tensorflow/datasets/blob/e2241839349317a6d38ea8198a44458ca412e7a6/tensorflow_datasets/testing/test_data/fake_examples/tf_flowers/flower_photos.zip) , but in the above log 

>path = '/tmpfs/src/github/datasets/tensorflow_datasets/testing/test_data/fake_examples/tf_flowers/flower_photos'

is not showing the changes.
It has to be 
```
path = '/tmpfs/src/github/datasets/tensorflow_datasets/testing/test_data/fake_examples/tf_flowers/flower_photos'.zip
```

File: tensorflow_datasets/image/flowers.py [hunk 1]
```diff
@@ -35,65 +36,71 @@
...
+                    names=["dandelion",
+                           "daisy",
+                           "tulips",
+                           "sunflowers",
+                           "roses",]
+                ),
+            }
+        ),
         supervised_keys=("image", "label"),
         urls=[_URL],
```
Comment (ChanchalKumarMaji) [inline tensorflow_datasets/image/flowers.py]: Should be replaced by `homepage=_URL`.

**Comment by Suhasnama** [comment]
@ChanchalKumarMaji , resolved the merge conflicts and modified `homepage=_URL`

**Comment by Suhasnama** [comment]
Hello @Conchylicultor , @cyfra , is this PR ready to merge?

File: tensorflow_datasets/image/flowers.py [hunk 2]
```diff
@@ -46,49 +51,50 @@ def _info(self):
     return tfds.core.DatasetInfo(
         builder=self,
         description="A large set of images of flowers",
-        features=tfds.features.FeaturesDict({
-            "image": tfds.features.Image(),
-            "label": tfds.features.ClassLabel(
-                names=["dandelion", "daisy", "tulips", "sunflowers", "roses"]),
-        }),
+        features=tfds.features.FeaturesDict(
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/flowers.py]: Why modifying the formatting here ?

File: tensorflow_datasets/image/flowers.py [hunk 3]
```diff
@@ -46,49 +51,50 @@ def _info(self):
...
-          for image_file in fname:
-            if image_file.endswith(".jpg"):
-              image_path = os.path.join(full_path, image_file)
-              record = {
-                  "image": image_path,
-                  "label": d.lower(),
-              }
-              yield "%s/%s" % (d, image_file), record
+    for fname, fobj in archive:
+      res = _NAME_RE.match(fname)
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/flowers.py]: I think regex are less readable, more error prone and overkill for this. I prefer the `image_file.endswith(".jpg")` check and the label could be extracted from the dir name (`fname.split('/')`[-2])

**Comment by Conchylicultor** [comment]
Closing this as https://github.com/tensorflow/datasets/pull/1929 was merged.

**Pull request closed without merging**